### ISSUE TITLE
Broken rule documentation links

### ISSUE THREAD

**Issue by jaydenseric** [xjamundx/eslint-plugin-promise#61]
Error tooltip rule documentation links in tools like [Atom linter-eslint](https://atom.io/packages/linter-eslint) are broken.

For example, `https://github.com/xjamundx/eslint-plugin-promise#catch-or-return` should be `https://github.com/xjamundx/eslint-plugin-promise#rule-catch-or-return`.

**Comment by macklinu** [comment]
Looking into this, it seems like Atom's `linter-eslint` package tries to determine a rule URL if one does not exist - check out [this code](https://github.com/AtomLinter/linter-eslint/blob/1427e52364351cbbe5ca433c23c6e8a23f72bfd3/src/rules.js#L39-L56).

One area of improvement for each rule in this plugin would be to refactor to include the [`meta.docs.url`](https://eslint.org/docs/developer-guide/working-with-rules#rule-basics) field. 

For example, the `catch-or-return` rule would go from looking like this:

```js
module.exports = {
  create: function (context) {
    // ...snippet
  }
}
```

To this:

```js
module.exports = {
  meta: {
    docs: {
      url: 'https://github.com/xjamundx/eslint-plugin-promise#rule-catch-or-return'
    }
  },
  create: function (context) {
    // ...snippet
  }
}
```

I am sure PRs are welcome for this contribution - can probably take a look this week, but feel free to comment if you'd like to contribute this, @jaydenseric. 🙂

**Comment by macklinu** [comment]
This wasn't resolved in #86 - mistyped in that PR description. Will resolve this issue in another PR.
